src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/tts_is_ready_request_test.cc

```diff
@@ -0,0 +1,306 @@
+      hmi_apis::FunctionID::TTS_GetCapabilities};
+  EXPECT_CALL(mock_hmi_capabilities_, GetInterfacesToUpdate())
+      .WillOnce(Return(interfaces_to_update));
+  ExpectSendMessagesToHMI();
+
+  ASSERT_TRUE(command_->Init());
+  command_->onTimeOut();
+}
+
+TEST_F(TTSIsReadyRequestTest,
+       OnEvent_TTSLanguageIsAbsentInCache_SendTTSGetLanguageRequest) {
+  std::vector<hmi_apis::FunctionID::eType> interfaces_to_update{
+      hmi_apis::FunctionID::TTS_GetLanguage};
+
+  Event event(hmi_apis::FunctionID::TTS_IsReady);
+
+  InterfacesUpdateExpectations(interfaces_to_update);
+
+  smart_objects::SmartObjectSPtr language(
+      std::make_shared<smart_objects::SmartObject>(
+          smart_objects::SmartType_Map));
+  (*language)[am::strings::params][am::strings::function_id] =
+      hmi_apis::FunctionID::TTS_GetLanguage;
+  EXPECT_CALL(mock_message_helper_,
+              CreateModuleInfoSO(hmi_apis::FunctionID::TTS_GetLanguage, _))
+      .WillOnce(Return(language));
+  EXPECT_CALL(mock_hmi_capabilities_, set_handle_response_for(*language));
+  EXPECT_CALL(mock_rpc_service_,
+              ManageHMICommand(
+                  HMIFunctionIDIs(hmi_apis::FunctionID::TTS_GetLanguage), _));
+
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageHMICommand(
+          HMIFunctionIDIs(hmi_apis::FunctionID::TTS_GetSupportedLanguages), _))
+      .Times(0);
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageHMICommand(
+          HMIFunctionIDIs(hmi_apis::FunctionID::TTS_GetCapabilities), _))
+      .Times(0);
+
+  ASSERT_TRUE(command_->Init());
+  command_->on_event(event);
+}
+
+TEST_F(
+    TTSIsReadyRequestTest,
+    OnEvent_TTSGetSupportedLanguagesIsAbsentInCache_SendTTSGetSupportedLanguagesRequest) {
+  std::vector<hmi_apis::FunctionID::eType> interfaces_to_update{
+      hmi_apis::FunctionID::TTS_GetSupportedLanguages};
+
+  Event event(hmi_apis::FunctionID::TTS_IsReady);
+
+  InterfacesUpdateExpectations(interfaces_to_update);
+  smart_objects::SmartObjectSPtr all_languages(
+      std::make_shared<smart_objects::SmartObject>(
+          smart_objects::SmartType_Map));
+  (*all_languages)[am::strings::params][am::strings::function_id] =
+      hmi_apis::FunctionID::TTS_GetSupportedLanguages;
+  EXPECT_CALL(
+      mock_message_helper_,
+      CreateModuleInfoSO(hmi_apis::FunctionID::TTS_GetSupportedLanguages, _))
+      .WillOnce(Return(all_languages));
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageHMICommand(
+          HMIFunctionIDIs(hmi_apis::FunctionID::TTS_GetSupportedLanguages), _));
+
+  EXPECT_CALL(mock_rpc_service_,
+              ManageHMICommand(
+                  HMIFunctionIDIs(hmi_apis::FunctionID::TTS_GetLanguage), _))
+      .Times(0);
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageHMICommand(
+          HMIFunctionIDIs(hmi_apis::FunctionID::TTS_GetCapabilities), _))
+      .Times(0);
+
+  ASSERT_TRUE(command_->Init());
+  command_->on_event(event);
```

Comment:
Ron not called. Is it safe to call on_event if Run was not called ?

Reply:
@LuxoftAKutsan Done in baa43da677085c27f015a706ddd5256b4a31f17e